src/codeu/chat/client/core/ConversationContext.java

```diff
@@ -56,7 +56,7 @@ public MessageContext add(String messageBody) {
 	
     return message == null ?
         null :
-        new MessageContext(message, view);
+        new MessageContext(message, conversation, user, view);
```

Comment:
Once again, you don't really need to add these new variables, both conversation and user the message belongs to are Uuid variables in the Message class. I understand you need both variables as non-Uuid parameters for authorityModificationRequest rn, but it's better to take them as Uuids in the method that you can grab a variable from later.